File: samples/jafu-servlet-cassandra/src/test/java/com/sample/IntegrationTests.java
package com.sample;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.test.web.reactive.server.WebTestClient;

import java.util.UUID;

class IntegrationTests {

    private final WebTestClient client = WebTestClient.bindToServer().baseUrl("http://localhost:8181").build();

    private ConfigurableApplicationContext context;

    @BeforeAll
    public void beforeAll() {
        Application.main(new String[]{"--spring.profiles.active=test"});
    }

    @Test
    void testSaveUser() {
        final User user = new User("Hamid", "M");

        client.post().uri("/users").bodyValue(user).exchange()
                .expectStatus().is2xxSuccessful()
                .expectBody()
                .jsonPath("$.firstName").isEqualTo("Hamid")
                .jsonPath("$.lastName").isEqualTo("M");
    }

    @Test
    void testFindAllUsers() {
        client.get().uri("/users").exchange()
                .expectStatus().is2xxSuccessful()
                .expectBodyList(User.class);
    }

    @Test
    void testFindOneUser() {
        client.get().uri("/users/" + UUID.randomUUID().toString()).exchange()
                .expectStatus().isNotFound();
    }

}

File: samples/graal-webapp/src/test/kotlin/org/springframework/fu/sample/graal/IntegrationTests.kt
package org.springframework.fu.sample.graal

import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.springframework.test.web.reactive.server.*
import org.springframework.test.web.reactive.server.WebTestClient

class IntegrationTests {

	private val client = WebTestClient.bindToServer().baseUrl("http://localhost:8080").build()

	@BeforeAll
	fun beforeAll() {
		app.run(profiles = "test")
	}

	@Test
	fun `Request root endpoint`() {
		client.get().uri("/").exchange()
				.expectStatus().is2xxSuccessful
				.expectBody<String>().isEqualTo("Hello world!")
	}

	@AfterAll
	fun afterAll() {
		app.stop()
	}
}